flink-runtime/src/main/java/org/apache/flink/runtime/state/StateBackend.java

```diff
@@ -201,4 +278,75 @@ default boolean supportsNoClaimRestoreMode() {
     default boolean supportsSavepointFormat(SavepointFormatType formatType) {
         return formatType == SavepointFormatType.CANONICAL;
     }
+
+    /**
+     * Parameters passed to {@link
+     * StateBackend#createKeyedStateBackend(KeyedStateBackendParameters)}.
+     *
+     * @param <K> The type of the keys by which the state is organized.
+     */
+    @PublicEvolving
+    interface KeyedStateBackendParameters<K> {
+        /** @return The runtime environment of the executing task. */
+        Environment getEnv();
+
+        JobID getJobID();
+
+        String getOperatorIdentifier();
+
+        TypeSerializer<K> getKeySerializer();
+
+        int getNumberOfKeyGroups();
+
+        /** @return Range of key-groups for which the to-be-created backend is responsible. */
+        KeyGroupRange getKeyGroupRange();
+
+        TaskKvStateRegistry getKvStateRegistry();
+
+        /** @return Provider for TTL logic to judge about state expiration. */
+        TtlTimeProvider getTtlTimeProvider();
+
+        MetricGroup getMetricGroup();
+
+        @Nonnull
+        Collection<KeyedStateHandle> getStateHandles();
+
+        /**
+         * @return The registry to which created closeable objects will be * registered during
+         *     restore.
+         */
+        CloseableRegistry getCancelStreamRegistry();
+
+        double getManagedMemoryFraction();
+
+        CustomInitializationMetrics getCustomInitializationMetrics();
+    }
+
+    /**
+     * Parameters passed to {@link
+     * StateBackend#createOperatorStateBackend(OperatorStateBackendParameters)}.
+     */
+    @PublicEvolving
+    interface OperatorStateBackendParameters {
+        /** @return The runtime environment of the executing task. */
+        Environment getEnv();
```

Comment:
As said in a previous comment, not a fan of the extra interfaces, I'd simple turn them into classes. And then they could have a common superclass of env, operator id, etc.